src/com/dotcms/api/system/user/UserServiceFactory.java

```diff
@@ -0,0 +1,366 @@
+			 *            include in the query result.
+			 * @param includeAnonymous
+			 *            - If set to {@code true}, the "Anonymous" user will be
+			 *            included in the result. Otherwise, set to
+			 *            {@code false}.
+			 * @param includeDefault
+			 *            - If set to {@code true}, the "Default" user will be
+			 *            included in the result. Otherwise, set to
+			 *            {@code false}.
+			 */
+			public UsersListTemplate(String inode, int permissionType, String filter, int start, int limit,
+					boolean includeAnonymous, boolean includeDefault) {
+				this.inode = inode;
+				this.permissionType = permissionType;
+				this.filter = filter;
+				this.start = start;
+				this.limit = limit;
+				this.includeAnonymous = includeAnonymous;
+				this.includeDefault = includeDefault;
+			}
+
+			/**
+			 * Executes this filtering template based on the specified search
+			 * criteria.
+			 * 
+			 * @return A {@link Map} with the result of the filtering query.
+			 */
+			public Map<String, Object> perform() {
+				ArrayList<Map<String, String>> list = null;
+				Map<String, Object> results = new HashMap<String, Object>(2);
+				int totalItemCount = 0;
+				List<User> users = null;
+				int realUserCount = 0;
+				try {
+					// Step 1. Retrieve users, beginning from "start" parameter,
+					// up to a number of "limit" items, filtered by "filter"
+					// parameter.
+					totalItemCount = getUserCount();
+					if (start < totalItemCount) {
+						users = getUsers();
+						realUserCount = users.size();
+					}
+					// Step 2. Assemble all of this information into an
+					// appropriate container to the view
+					if (users != null) {
+						int pageSize = realUserCount;
+						list = new ArrayList<Map<String, String>>(pageSize);
+						for (User aUser : users) {
+							Map<String, String> aRecord = new HashMap<String, String>();
+							String fullName = aUser.getFullName();
+							fullName = (UtilMethods.isSet(fullName) ? fullName : " ");
+							String emailAddress = aUser.getEmailAddress();
+							emailAddress = (UtilMethods.isSet(emailAddress) ? emailAddress : " ");
+							aRecord.put("id", aUser.getUserId());
+							aRecord.put("type", USER_TYPE_VALUE);
+							aRecord.put("name", fullName);
+							aRecord.put("emailaddress", emailAddress);
+							list.add(aRecord);
+						}
+					} else {
+						list = new ArrayList<Map<String, String>>(0);
+					}
+				} catch (Exception ex) {
+					Logger.warn(UserAjax.class, "::processUsersList -> Could not process list of users.");
+					list = new ArrayList<Map<String, String>>(0);
+				}
+				results.put("data", list);
+				results.put("total", totalItemCount);
+				return results;
+			}
+		}
+
+		@Override
+		public Map<String, Object> getUsersList(String assetInode, String permission, Map<String, String> params)
+				throws Exception {
+			int start = 0;
+			if (params.containsKey("start")) {
```

Comment:
You might want to include this parsing into ConversionUtils and handle the NumberException, etc.
Also you can create encapsulate in a method, something like:

int getParam(final String name, Map<String, String> params) {

   int result = -1;

  if(params.containsKey(name)) {
    result = ConversionUtils.toInteger (params.get(name));
 } 

 return result;
}